### PR TITLE
Port sanity tests to v2

### PR DESCRIPTION
**Pull request by sirutBuasai**
## Purpose

This migrate sanity tests from v1 to v2. 
PR does the following:
- Deprecate old tests that are no longer applicable
- Remove tests that are framework specific, sanity tests should be run for all images. Framework specfic tests should live in their own framework tests directory.

### V2 sanity tests consists of the following
- Checkout + ECR authenticate + pull image
- Setup host test dependencies (uv venv + pip install)
- Start persistent container (mounts repo at /workdir)
- **Filesystem checks** (clean image state, before any pip install):
   - Check stray .py artifacts
   - Check /tmp and /var/tmp
   - Check cache dir
   - Check no viminfo/bash history
   - Check no anaconda packages
- **Version checks**:
   - Check Python version
   - Check OS version
   - Check framework version (skipped for base)
- **Label checks** (host-side):
   - Check container labels (URI visibility + length)
- **OSS compliance**:
   - Run testOSSCompliance in container
   - docker cp THIRD_PARTY_SOURCE_CODE_URLS
   - Upload source tarballs to S3
- **Package checks** (after pip install boto3):
   - Setup: pip install boto3 botocore in container
   - Run pip check with allowlists

### The following tests are dropped:
| Test | File | Reason |
|------|------|--------|
| `test_health_check` | `test_health_check.py` | Permanently skipped — EFA binaries not maintained by DLC |
| `test_mxnet_blas_mkl` | `test_mxnet_blas_mkl.py` | MXNet deprecated by AWS |
| `test_safety_check` | `test_safety_check.py` | File header says no longer used; superseded by ECR scan |
| `test_pt_s3` | `test_pt_s3.py` | Only for PyTorch 1.8-1.12; v2 ships PT ≥ 2.x |
| `test_torchvision` | `test_torchvision.py` | Skips for PT ≥ 1.10; dead code in v2 |
| `test_canary_integration` | `test_canary_integration.py` | Deep canary context only; not applicable to v2 |
| `test_dataclasses_check` | `test_pre_release.py` | Niche check for Python < 3.7; v2 is 3.7+ only |
| `test_image_tags` | `test_pre_release.py` | ECR tag check; not needed in v2 |
| `test_core_package_version` | `test_pre_release.py` | Remnant of autopatch system not used in v2 |
| `test_package_version_regression` | `test_pre_release.py` | Compares against previous release; dropped due to autopatch deprecation |
| `test_tf_serving_version_cpu` | `test_pre_release.py` | TF-specific; sanity tests should be universal |
| `test_tf_serving_api_version` | `test_pre_release.py` | TF-specific |
| `test_sm_toolkit_and_ts_version_*` | `test_pre_release.py` | Framework-specific SM toolkit checks |
| `test_framework_and_neuron_sdk_version` | `test_pre_release.py` | Neuron-specific |
| `test_*_sm_env_variables` (4 tests) | `test_pre_release.py` | Framework-specific env var checks |
| `test_mxnet_training_sm_env_variables` | `test_pre_release.py` | MXNet deprecated |
| `test_awscli` | `test_utility_installation.py` | MXNet-only |
| `test_boto3` | `test_utility_installation.py` | MXNet-only |
| `test_utility_packages_using_import` | `test_utility_installation.py` | Framework-specific |
| `test_common_pytorch_utility_packages` | `test_utility_installation.py` | PyTorch-specific |
| `test_mpi4py_for_*` (3 tests) | `test_utility_installation.py` | Framework-specific |
| `test_sagemaker_studio_analytics_extension` | `test_utility_installation.py` | Old PT/TF versions only |
| `test_ipykernel_presence` | `test_utility_installation.py` | TF-specific |
| `quick_checks/test_buildspecs.py` | `quick_checks/` | Not used in v2 |
| `quick_checks/test_dlc_developer_config.py` | `quick_checks/` | Not used in v2 |

The following tests are deferred to another PR due to scope
### Deferred — CVE/Security Scanning (port later)
| Test | File | Reason |
|------|------|--------|
| `test_ecr_scan` | `test_ecr_scan.py` | Heavy dependency on test_utils/security.py; ECR scan classes |
| `test_safety_report_file` | `test_safety_report_file.py` | Related to safety/CVE scanning |
| `test_scan_vulnerability_class_functionality` | `test_scan_vulnerability_class_functionality.py` | Unit tests for ECR scan classes |

### Deferred — Other
| Test | File | Reason |
|------|------|--------|
| `test_telemetry` | `test_telemetry.py` | Deferred; checks PyTorch telemetry env vars |
| `test_cuda_paths` | `test_pre_release.py` | Host-side buildspec/Dockerfile path validation |
| `test_license_file` (full) | `test_pre_release.py` | Full S3 comparison deferred; TODO in workflow to add license check for sglang |

 

## Test Result

______________________________________________________________________

<details>
<summary>Toggle if you are merging into master Branch</summary>

By default, docker image builds and tests are disabled. Two ways to run builds and tests:

1. Using dlc_developer_config.toml
1. Using this PR description (currently only supported for PyTorch, TensorFlow, vllm, and base images)

<details>
<summary>How to use the helper utility for updating dlc_developer_config.toml</summary>

Assuming your remote is called `origin` (you can find out more with `git remote -v`)...

- Run default builds and tests for a particular buildspec - also commits and pushes changes to remote; Example:

`python src/prepare_dlc_dev_environment.py -b </path/to/buildspec.yml> -cp origin`

- Enable specific tests for a buildspec or set of buildspecs - also commits and pushes changes to remote; Example:

`python src/prepare_dlc_dev_environment.py -b </path/to/buildspec.yml> -t sanity_tests -cp origin`

- Restore TOML file when ready to merge

`python src/prepare_dlc_dev_environment.py -rcp origin`

**NOTE: If you are creating a PR for a new framework version, please ensure success of the local, standard, rc, and efa sagemaker tests by updating the dlc_developer_config.toml file:**

- [ ] `sagemaker_remote_tests = true`
- [ ] `sagemaker_efa_tests = true`
- [ ] `sagemaker_rc_tests = true`
- [ ] `sagemaker_local_tests = true`

</details>

<details>
<summary>How to use PR description</summary>
Use the code block below to uncomment commands and run the PR CodeBuild jobs. There are two commands available:

- `# /buildspec <buildspec_path>`
  - e.g.: `# /buildspec pytorch/training/buildspec.yml`
  - If this line is commented out, dlc_developer_config.toml will be used.
- `# /tests <test_list>`
  - e.g.: `# /tests sanity security ec2`
  - If this line is commented out, it will run the default set of tests (same as the defaults in dlc_developer_config.toml): `sanity, security, ec2, ecs, eks, sagemaker, sagemaker-local`.

</details>

```
# /buildspec <buildspec_path>
# /tests <test_list>
```

</details>

<details>
<summary>Toggle if you are merging into main Branch</summary>

## PR Checklist

- [] I ran `pre-commit run --all-files` locally before creating this PR. (Read [DEVELOPMENT.md](https://github.com/aws/deep-learning-containers/blob/main/DEVELOPMENT.md) for details).

</details>

### CODE REVIEW

File: test/sanity/scripts/check_framework_version.sh [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#!/usr/bin/env bash
```
Comment (junpuf) [inline test/sanity/scripts/check_framework_version.sh]: let's get rid of this file, this is meaningless

File: test/sanity/scripts/check_cuda.sh [hunk 2]
```diff
@@ -0,0 +1,27 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+# Check CUDA version matches expected
```
Comment (junpuf) [inline test/sanity/scripts/check_cuda.sh]: drop this one too

File: test/sanity/scripts/check_os_version.sh [hunk 3]
```diff
@@ -0,0 +1,16 @@
+#!/usr/bin/env bash
+set -euo pipefail
+EXPECTED="${1:?Usage: check_os_version.sh <os-version>}"
```
Comment (junpuf) [inline test/sanity/scripts/check_os_version.sh]: drop this too

File: test/sanity/scripts/check_python_version.sh [hunk 4]
```diff
@@ -0,0 +1,12 @@
+#!/usr/bin/env bash
```
Comment (junpuf) [inline test/sanity/scripts/check_python_version.sh]: drop this too

File: test/sanity/scripts/pip_check.py [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+"""Pip check with 3-level allowlist support.
```
Comment (junpuf) [inline test/sanity/scripts/pip_check.py]: drop this too
